pkg/apis/core/validation/shoot.go

```diff
@@ -63,6 +63,24 @@ var (
 		string(corev1.ServiceExternalTrafficPolicyTypeCluster),
 		string(corev1.ServiceExternalTrafficPolicyTypeLocal),
 	)
+	availableShootOperations = sets.NewString(
```

Comment:
From this, I can understand that maintenance operation doesn't support retry and maintenance. But [here](https://github.com/gardener/gardener/blob/b877ebe0c96846917595a728ac2be71bff6a84d4/docs/usage/shoot_maintenance.md?plain=1#L81) we suggest we can use any shoot operation. Probably we should improve this documentation also.